static/src/javascripts/projects/common/modules/identity/upsell/account-creation/AccountCreationForm.js

```diff
@@ -0,0 +1,166 @@
+// @flow
+import React, { Component } from 'preact-compat';
+import reqwest from 'reqwest';
+import ophan from 'ophan/ng';
+import reportError from 'lib/report-error';
+import { AccountBenefits } from './AccountBenefits';
+
+type AccountCreationFormProps = {
+    csrfToken: string,
+    accountToken: string,
+    email: string,
+    onAccountCreated: () => {},
+};
+
+class AccountCreationForm extends Component<
+    AccountCreationFormProps,
+    {
+        password?: string,
+        isLoading?: boolean,
+        isError?: boolean,
+        errorReason?: string,
+    }
+> {
+    onSubmit = (ev: Event) => {
+        ev.preventDefault();
+        this.setState({
+            isLoading: true,
+            isError: false,
+        });
+
+        reqwest({
+            url: '/password/guest',
+            method: 'post',
+            data: {
+                csrfToken: this.props.csrfToken,
+                token: this.props.accountToken,
+                password: this.state.password,
+            },
+            success: () => {
+                ophan.record({
+                    component: 'set-password',
+                    value: 'set-password',
+                });
+                this.props.onAccountCreated();
+            },
+            error: response => {
+                reportError(Error(response), {
+                    feature: 'identity-create-account-upsell',
+                });
+                try {
+                    const apiError = JSON.parse(response.responseText)[0];
+                    this.setState({
+                        isError: true,
+                        errorReason: apiError.description,
+                    });
+                } catch (exception) {
+                    this.setState({ isError: true });
+                }
+            },
+            complete: () => {
+                this.setState({ isLoading: false });
+            },
+        });
+    };
+
+    handlePasswordChange = (ev: Event) => {
+        if (!(ev.target instanceof HTMLInputElement)) {
+            return;
+        }
+        this.setState({ password: ev.target.value });
+    };
+
+    render() {
+        const { isError, errorReason, isLoading } = this.state;
+        const { email } = this.props;
+        return (
+            <form className="form" onSubmit={this.onSubmit}>
+                <hr className="manage-account-small-divider" />
+                {isError && (
+                    <div className="form__error">
+                        {errorReason || 'Oops. Something went wrong'}
+                    </div>
+                )}
+                <h1 className="identity-upsell-title">
+                    <h1 className="identity-upsell-title__title">
+                        Want more from The Guardian?
+                    </h1>
+                    <p className="identity-upsell-title__subtitle">
+                        Create your account now to manage your preferences and
```

Comment:
Is this message a bit misleading, given they already have a (soft) account at this point?